### PR TITLE
Do not add empty lines when attempting to preserve formatting that ap…cd 

### PR DESCRIPTION
**Pull request by pranavkm**
…pears prior to clear elements

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @yishaigalatzer @zhili1208 @deepakaravindr

File: test/NuGet.Configuration.Test/PackageSourceProviderTests.cs [hunk 1]
```diff
@@ -798,19 +796,17 @@ public void SavePackageSourcesWithOnlyClear()
...
 <configuration>
     <packageSources>
-        <clear /> 
+        <clear />
         <add key=""test3"" value=""https://test3.net"" />
     </packageSources>
     <disabledPackageSources>
-        <clear />
-    </disabledPackageSources>
+        <clear /></disabledPackageSources>
```
Comment (pranavkm) [inline test/NuGet.Configuration.Test/PackageSourceProviderTests.cs]: It ends up looking messed up in this scenario (the `clear` element is the last element) - I guess we can special case for this if it's pertinent?

File: src/NuGet.Configuration/Settings/Settings.cs [hunk 2]
```diff
@@ -496,20 +498,28 @@ private static void RemoveElementAfterClearTag(XElement sectionElement)
...
-            var removeElmentList = new List<XElement>();
-            foreach (var element in sectionElement.Elements())
+            var nodesToRemove = new List<XNode>();
+            foreach (var node in sectionElement.Nodes())
             {
+                if (node.NodeType != XmlNodeType.Element)
+                {
+                    nodesToRemove.Add(node);
+                    continue;
+                }
```
Comment (yishaigalatzer) [inline src/NuGet.Configuration/Settings/Settings.cs:508]: nice use of continue

File: test/NuGet.Configuration.Test/PackageSourceProviderTests.cs (hunk 1 above)
Comment (yishaigalatzer) [inline test/NuGet.Configuration.Test/PackageSourceProviderTests.cs]: that would be nice, but not critical

**Comment by yishaigalatzer** [comment]
:shipit: after you make a decision about the </clear> as last element. I don't think its really common or ever a thing to have clear as last element, but this is the same behavior for sources we should fix

**Comment by pranavkm** [comment]
Updated with a fix for `clear` being the last node.

**Comment by pranavkm** [comment]
cc @danliu

**Comment by pranavkm** [comment]
Merged in https://github.com/NuGet/NuGet3/commit/16327a708887aabdfc20d02631897a8a731ab1bf

**Pull request closed without merging**